### ISSUE TITLE
Only most recent Eval is viewable in the WebViewer - Failed to execute 'setItem' on 'Storage'

### ISSUE THREAD

**Issue by CMelnickEngageware** [promptfoo/promptfoo#802]
# Description
Only the most recent evaluation is able to be loaded in the WebViewer.  Each time a new eval is run, any previous evals that have been run do not load in the 'Evals' tab of the WebViewer.

## Steps to reproduce:
1.  Run an eval
2. View it in the 'Evals' tab of the WebViewer
3. Run a second eval
4. Attempt to load the eval from step 1 by selecting it in the 'Progress' tab

**NOTE:** the size of the eval data (prompt, context, response, # of test cases) may be the issue.  I am running large evals comparing 3 different models

<img width="1720" alt="Screenshot 2024-05-20 at 12 33 52 PM" src="https://github.com/promptfoo/promptfoo/assets/95300041/13cc520c-e8af-45bd-8904-3b4730ab1079">

## Additional Context
Google Chrome (Version 124.0.6367.91 (Official Build) (arm64)) on an M1 Macbook Pro.

Error in console of Chrome:

> Uncaught (in promise) DOMException: Failed to execute 'setItem' on 'Storage': Setting the value of 'ResultsViewStorage' exceeded the quota.

I believe this issue originates from the code within `web/nextui/src/app/eval/store.ts`:
```typescript
export const useStore = create<TableState>()(
  persist(
    (set, get) => ({
      evalId: null,
      setEvalId: (evalId: string) => set(() => ({ evalId })),
      ........
    }),
    {
      name: 'ResultsViewStorage',
      storage: createJSONStorage(() => localStorage),
    },
  ),
);
```

**Comment by typpo** [comment]
Are your evals particularly large and/or do you have a lot of them?

**Comment by CMelnickEngageware** [comment]
I am running 130 evals over 3 different providers to compare 3 different OpenAI models.  Each eval contains a fairly large context window.  To answer your question more directly - yes, I am running large evals and I am running a lot of them.  Would something like IndexedDB be more suited or is the WebViewer not meant to support this large of a test suite?  I would be willing to help fix if need be.

**Comment by typpo** [comment]
I would love for the viewer to support this use case... IndexedDB is
probably a solution that makes sense here!

**Comment by CMelnickEngageware** [comment]
I would be happy to submit a PR for this.  I quickly tried getting everything running locally this morning but ran out of time.  It looks like the development documentation is a little out of date.  Would you mind updating?

**Comment by typpo** [comment]
I went ahead and included this in 0.63.0 (I think the dev documentation has been updated as well)
